internal/errors/grpc_test.go

```diff
@@ -0,0 +1,555 @@
+package errors
+
+import (
+	"math"
+	"testing"
+)
+
+func TestErrGRPCClientConnectionClose(t *testing.T) {
+	type args struct {
+		name string
+		err  error
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return wrapped ErrGRPCClientConnectionClose error when err is server error and name is 'gateway'",
+			args: args{
+				err:  New("server error"),
+				name: "gateway",
+			},
+			want: want{
+				want: New("gateway's gRPC connection close error: server error"),
+			},
+		},
+		{
+			name: "return wrapped ErrGRPCClientConnectionClose error when err is server error and name is empty",
+			args: args{
+				err:  New("server error"),
+				name: "",
+			},
+			want: want{
+				want: New("'s gRPC connection close error: server error"),
+			},
+		},
+		{
+			name: "return ErrGRPCClientConnectionClose error when err is nil error and name is 'gateway'",
+			args: args{
+				err:  nil,
+				name: "gateway",
+			},
+			want: want{
+				want: New("gateway's gRPC connection close error"),
+			},
+		},
+		{
+			name: "return ErrGRPCClientConnectionClose error when err is nil error and addr is empty",
+			args: args{
+				err:  nil,
+				name: "",
+			},
+			want: want{
+				want: New("'s gRPC connection close error"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrGRPCClientConnectionClose(test.args.name, test.args.err)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrInvalidGRPCPort(t *testing.T) {
+	type args struct {
+		addr string
+		host string
+		port uint16
```

Comment:
I thinks it'd be better add case with boundary value of `uint16`

Reply:
Thank you for your comment. :bow:
I agree with you. 👍 


I selected `0` and `maxUint16`and `8080` as port value. 🤔
Originally, `maxUint16 + 1`, `-1` should be also included as boundary value, but it could not be included due to a compile error. ...
So could you tell me which value should be selected as the boundary value of uint16?
sorry 🙏

Reply:
For example, uint16 range is from `0` to `maxUint16 ` 🤔 
so, is the boundary value below?
- `0`
- `1`
- `maxUint16-1`
- `maxUint16`

Reply:
yap

Reply:
plz refer to https://golang.org/pkg/math/#pkg-constants

Reply:
@vankichi 
thank you